bindings/go/helm/cmd/main.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"log/slog"
+	"os"
+
+	"ocm.software/open-component-model/bindings/go/blob"
+	filesystemv1alpha1 "ocm.software/open-component-model/bindings/go/configuration/filesystem/v1alpha1/spec"
+	genericv1 "ocm.software/open-component-model/bindings/go/configuration/generic/v1/spec"
+	constructorruntime "ocm.software/open-component-model/bindings/go/constructor/runtime"
+	helminput "ocm.software/open-component-model/bindings/go/helm/input"
+	helmv1 "ocm.software/open-component-model/bindings/go/helm/input/spec/v1"
+	plugin "ocm.software/open-component-model/bindings/go/plugin/client/sdk"
+	"ocm.software/open-component-model/bindings/go/plugin/manager/contracts"
+	v1 "ocm.software/open-component-model/bindings/go/plugin/manager/contracts/input/v1"
+	"ocm.software/open-component-model/bindings/go/plugin/manager/endpoints"
+	"ocm.software/open-component-model/bindings/go/plugin/manager/registries/input"
+	"ocm.software/open-component-model/bindings/go/plugin/manager/types"
+	"ocm.software/open-component-model/bindings/go/runtime"
+)
+
+// HelmInputPlugin is a plugin that implements the helm.InputMethod interface as an external binary.
+type HelmInputPlugin struct {
+	contracts.EmptyBasePlugin
+	filesystemConfig *filesystemv1alpha1.Config
+}
+
+var logger *slog.Logger
+
+func (h *HelmInputPlugin) GetIdentity(ctx context.Context, typ *v1.GetIdentityRequest[runtime.Typed]) (*v1.GetIdentityResponse, error) {
+	logger.Info("GetIdentity called for Helm input", "type", typ.Typ)
+	return nil, nil
+}
+
+func (h *HelmInputPlugin) ProcessResource(ctx context.Context, request *v1.ProcessResourceInputRequest, credentials map[string]string) (*v1.ProcessResourceInputResponse, error) {
+	logger.Info("ProcessResource called for Helm input")
+	return processHelmResource(ctx, request, credentials, h.filesystemConfig)
+}
+
+func (h *HelmInputPlugin) ProcessSource(ctx context.Context, request *v1.ProcessSourceInputRequest, credentials map[string]string) (*v1.ProcessSourceInputResponse, error) {
+	return nil, fmt.Errorf("not implemented")
+}
+
+var _ v1.ResourceInputPluginContract = &HelmInputPlugin{}
+
+func main() {
+	args := os.Args[1:]
+	logger = slog.New(slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{
+		Level: slog.LevelDebug,
+	}))
+
+	capabilities := endpoints.NewEndpoints(helminput.Scheme)
+
+	if len(args) > 0 && args[0] == "capabilities" {
+		// Register a temporary plugin instance for capabilities
+		if err := input.RegisterInputProcessor(&helmv1.Helm{}, &HelmInputPlugin{}, capabilities); err != nil {
```

Comment:
this feels off (having this temporary registration) cant we just always register it?

Reply:
Ah good catch. I think this is wrong indeed.

Reply:
Ah I remember now. I thought it's a cool "performance" thing that I register it temporarily to have it return immediately, but then do the real thing once it's starting. It's stupid, I'll remove it.